GANDLF/compute/training_loop.py

```diff
@@ -217,24 +215,24 @@ def train_network(model, train_dataloader, optimizer, params):
 
 
 def training_loop(
-    training_data,
-    validation_data,
-    device,
-    params,
-    output_dir,
-    testing_data=None,
-    epochs=None,
-):
+    training_data: pd.DataFrame,
+    validation_data: pd.DataFrame,
+    device: str,
+    params: dict,
+    output_dir: str,
+    testing_data: bool = None,
```

Comment:
testing_data should be of type Union[pd.DataFrame, None] according to the presented docstring and default value.

Reply:
For optional arguments, I propose to use the [`Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) identifier for clarity.